cmd/snap/cmd_info.go

```diff
@@ -300,7 +300,11 @@ func (x *infoCmd) Execute([]string) error {
 		both := coalesce(local, remote)
 
 		if both == nil {
-			fmt.Fprintf(w, "argument:\t%q\nwarning:\t%s\n", snapName, i18n.G("not a valid snap"))
+			if len(x.Positional.Snaps) == 1 {
+				return fmt.Errorf("no snap named %q found", snapName)
```

Comment:
as the string given can be a path as well as a name, wouldn't something like `"no snap found for %q"` work better here?